### PR TITLE
SRCH-104 fix SearchgovDomain#domain validation

### PR DESCRIPTION
**Pull request by MothOnMars**
SRCH-104 fix SearchgovDomain#domain validation
- remove explicit dependency on PublicSuffix gem
- clean up SearchgovDomain code per Rubocop

### CODE REVIEW

**Comment by MothOnMars** [comment]
@peggles2 , this is ready for review. The two codeclimate issues will be resolved once https://github.com/GSA/search-gov/pull/242 , which includes the Rubocop line limit increase, is merge to master.

File: app/models/searchgov_domain.rb [hunk 1]
```diff
@@ -66,29 +71,26 @@ def check_status
   def sitemap_urls
     urls = sitemaps.pluck(:url)
     urls += robotex.sitemaps(url).uniq.
-             reject{ |url| URI(url).host != domain }.
-             map{ |url| UrlParser.update_scheme(url, scheme) }
+              select { |url| URI(url).host == domain }.
+              map { |url| UrlParser.update_scheme(url, scheme) }
     urls.presence || ["#{url}sitemap.xml"]
```
[the remark this answers is not in the document]
Comment (MothOnMars) [reply app/models/searchgov_domain.rb]: Scopes are for ActiveRecord searches. In this case, the values are being pulled from a `robots.txt` file.

[the remark this answers is not in the document]
Comment (peggles2) [reply app/models/searchgov_domain.rb]: 👍

[the remark this answers is not in the document]
Comment (peggles2) [reply app/models/searchgov_domain.rb]: Once the code climate errors are fixed LGTM 👍

**Pull request merged**